### PR TITLE
🔧 Add imports order rule

### PR DESCRIPTION
**Pull request by XavierGuilloton**

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -59,6 +59,23 @@ module.exports = {
     // Prefer named export
     'import/prefer-default-export': 'off',
     'import/no-default-export': 'error',
+
+    // Order import with line separation between groups
+    'import/order': [
+      'error',
```
Comment (tgerboui) [inline index.js:65]: `warning` for now, this is not obligation for me.
See to set `error` in another major version.

File: index.js [hunk 2]
```diff
@@ -59,6 +59,23 @@ module.exports = {
     // Prefer named export
     'import/prefer-default-export': 'off',
     'import/no-default-export': 'error',
+
+    // Order import with line separation between groups
```
Comment (tgerboui) [inline index.js:63]: Ajouter aussi de la doc dans le README.

Comment (tgerboui) [inline index.js:63]: Add also documentation in `README.md`.

File: index.js [hunk 3]
```diff
@@ -59,6 +59,23 @@ module.exports = {
...
+    'import/order': [
+      'error',
+      {
+        // Add module import (@Module) after external import
```
Comment (tgerboui) [inline index.js:67]: Do not relate to business logic (`module`) here, module is only sensfull in some projects.

File: index.js [hunk 4]
```diff
@@ -59,6 +59,23 @@ module.exports = {
...
+        pathGroups: [
+          {
+            pattern: '@*',
```
Comment (tgerboui) [inline index.js:70]: Be careful here with this pattern.
Some package like for example `@nestjs/common` use '@' at begining which is not a locale directory.

Maybe target just only local file if possbile of new line required.(also regular import)

**Review by tgerboui — CHANGES_REQUESTED** [review]

**Comment by tgerboui** [comment]
And I had many strange errors when testing this import rule, did you try it ?

Errors like this when I have only one local import :
Screenshot 2022-04-19 at 19 07 49

**Comment by wayglem** [comment]
Closing as this is too old

**Pull request closed without merging**